### PR TITLE
Deploy contracts to different networks

### PR DESCRIPTION
**Pull request by edakturk14**
fixes https://github.com/scaffold-eth/se-2/issues/20

This is the first iteration to deploy the contract to other networks. We have a default private key and a method to generate a new mnenoic to have default settings for deployment.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,5 +22,8 @@
   "devDependencies": {
     "husky": "^8.0.1",
     "lint-staged": "^13.0.3"
+  },
+  "dependencies": {
+    "dotenv": "^16.0.3"
```
Comment (carletex) [inline package.json:27]: This should be installed inside the hardhat folder. 

This is a monorepo and we should see each folder a different projects / repo.

File: packages/hardhat/hardhat.config.ts [hunk 2]
```diff
@@ -10,6 +17,29 @@ const config: HardhatUserConfig = {
       default: 0,
     },
   },
+  networks: {
+    // order by abc
```
Comment (carletex) [inline packages/hardhat/hardhat.config.ts:21]: Is this a ToDo?

File: packages/hardhat/.env.example [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ALCHEMY_API_KEY=""
+DEPLOYER_PRIVATE_KEY=""
```
Comment (carletex) [inline packages/hardhat/.env.example:2]: Maybe we could a line on the README to tell people to copy this to `.env` and fill with their keys?

File: packages/hardhat/hardhat.config.ts [hunk 4]
```diff
@@ -1,7 +1,14 @@
+import * as dotenv from "dotenv";
+dotenv.config();
 import { HardhatUserConfig } from "hardhat/config";
 import "@nomicfoundation/hardhat-toolbox";
 import "hardhat-deploy";
 
+// ToDo: add default api key
+const defaultApiKey = process.env.ALCHEMY_API_KEY ?? '';
+// ToDo: auto generate mnenoic for the deployer account
+const deployerPrivateKey = process.env.DEPLOYER_PRIVATE_KEY ?? '';
```
Comment (carletex) [inline packages/hardhat/hardhat.config.ts:10]: '' => ""

We definitely need to have linting (with prettier) on the hardhat folder also. We could create a issue for it.

**Review by carletex — CHANGES_REQUESTED** [review]
Hey @edakturk14 

Thanks for this. Agree that this is a good first iteration. I saw that you created #25 for next steps. Great!

I added some review comments.

File: packages/hardhat/hardhat.config.ts (hunk 2 above)
Comment (edakturk14) [reply packages/hardhat/hardhat.config.ts]: Thanks!! yes was an internal note, fixed here https://github.com/scaffold-eth/se-2/pull/24/commits/a4ff6938c1d9b824909ed9f407a5a200f07d2486

File: package.json (hunk 1 above)
Comment (edakturk14) [reply package.json]: fixes here https://github.com/scaffold-eth/se-2/pull/24/commits/0fc68d94d6087a33f827eec86a1f74aa80441e73

File: packages/hardhat/hardhat.config.ts (hunk 4 above)
Comment (edakturk14) [reply packages/hardhat/hardhat.config.ts:10]: created new issue https://github.com/scaffold-eth/se-2/issues/26

**Comment by edakturk14** [comment]
made the changes!

File: packages/hardhat/hardhat.config.ts [hunk 5]
```diff
@@ -10,6 +17,40 @@ const config: HardhatUserConfig = {
       default: 0,
     },
   },
+  networks: {
+    arbiturum: {
```
Comment (carletex) [inline packages/hardhat/hardhat.config.ts:21]: There is a typo here. "arbiturum" => "arbitrum"

File: packages/hardhat/hardhat.config.ts [hunk 6]
```diff
@@ -10,6 +17,40 @@ const config: HardhatUserConfig = {
...
+      url: `https://arb-mainnet.g.alchemy.com/v2/${defaultApiKey}`,
+      accounts: [deployerPrivateKey]
+    },
+    arbiturumTestnet: {
```
Comment (carletex) [inline packages/hardhat/hardhat.config.ts:25]: same  "arbiturum" typo here.

File: packages/hardhat/hardhat.config.ts [hunk 7]
```diff
@@ -10,6 +17,40 @@ const config: HardhatUserConfig = {
...
+      url: `https://arb-rinkeby.g.alchemy.com/v2/${defaultApiKey}`,
```
Comment (carletex) [inline packages/hardhat/hardhat.config.ts:26]: Arbitrum rinkeby was deprecated after the merge. Let's switch to arbitrumGoerli. We can also change the name: arbitrumTestnet => arbitrumGoerli

**Review by carletex — CHANGES_REQUESTED** [review]
Thanks!! Left some review comments.

File: packages/hardhat/hardhat.config.ts (hunk 5 above)
Comment (edakturk14) [reply packages/hardhat/hardhat.config.ts]: changed here https://github.com/scaffold-eth/se-2/pull/24/commits/edeadba1050c1bbf017050d127156399364fe3e4

File: packages/hardhat/hardhat.config.ts (hunk 7 above)
Comment (edakturk14) [reply packages/hardhat/hardhat.config.ts]: fixed here https://github.com/scaffold-eth/se-2/pull/24/commits/634dc5c58f8a6ea6f51bf6bc7e7389ad11edd734

**Comment by edakturk14** [comment]
done!

**Review by carletex — APPROVED** [review]
Thanks, lgtm!

**Pull request merged**